### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by glrf**
This is my first attempt at a config package. I'm not quite sure about naming and structure I'm happy about any input.

The idea is that we have a `common.yml` as the base class. This base class contains both the `pkg.monitoring` default parameters (i.e. `class/defaults.yml`) and the base prometheus config that is relevant for all distributions.

We then have `openshift4.yml` as the Openshift specific class that includes `common` and adds Openshift specific config. We can add such classes for other distributions.

In practice to enable this package on an Openshift cluster we would import the `monitoring.openshift4` class and configure the necessary keys in `pkg.monitoring`. See the config on the beta cluster: [tenant-repo(internal)](https://git.vshn.net/syn-dev/customers/appuio/-/blob/7c1ac5ab6dcd31500296158fb4f4b562af7a77b1/c-old-fire-5788.yml#L451-L465)

I did not yet add documentation as I first wanted some input on the approach. We will probably want something like a `parameters.adoc` and some how-to that explains how to connect to keycloak and S3. I will add this in a separate PR when we are happy with the implementation

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: openshift4.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+classes:
+  - .common
+
+applications:
+  - prometheus
+
```
Comment (simu) [inline openshift4.yml]: This is not necessary, since we already add application `prometheus`in `common.yml`.

**Review by simu — APPROVED** [review]
LGTM overall.

I have no strong opinions regarding the package "parameters" layout, I think the current approach is as good as any, until we've gathered some experience.

File: common.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        tag: v0.24.0
+    prometheus:
+      ingress:
+        host: prometheus.example.com
+        annotations:
+          cert-manager.io/cluster-issuer: letsencrypt-production
+      proxy:
+        provider: keycloak-oidc
+        provider-display-name: "Test Account"
+        client-id: syn-monitoring-${cluster:name}
```
Comment (bastjan) [inline common.yml]: This will most likely change. We should speak with Dani before inventing client names.

As a default ok.

**Review by bastjan — APPROVED** [review]

File: common.yml (hunk 2 above)
Comment (glrf) [reply common.yml]: Fair enough. I think he likes underscores? :sweat_smile: 
We definitely should talk to him about naming as soon as we put this in production. I just chose some arbitrary default client id for now.

**Pull request merged**